NEXT_CHANGELOG.md

```diff
@@ -27,6 +27,24 @@ By [@USERNAME](https://github.com/USERNAME) in https://github.com/apollographql/
 ## ❗ BREAKING ❗
 ## 🚀 Features
 
+### Add configuration for trace ID ([Issue #2080](https://github.com/apollographql/router/issues/2080))
+
+If you want to expose in response headers the generated trace ID or the one you provided using propagation headers you can use this configuration:
+
+```yaml title="router.yaml"
+telemetry:
+  tracing:
+    experimental_expose_trace_id:
+      enabled: true # default: false
+      header_name: "my-trace-id" # default: "apollo-trace-id"
+    propagation:
+      custom_header: "x-request-id" # Specify your own trace_id with a custom header in request headers
```

Comment:
`custom_header` is inside `propagation` then it's used to set/propagate a trace id coming from the request headers.

`header_name` is inside `experimental_expose_trace_id` so it's the name of the response header to expose the trace_id. What could I do to improve this ? Add `response` and `request` in description maybe ?

Reply:
Suggest that `header_name` is renamed to `response_header_name`

Reply:
Ignoring the name of `header_name` **within** the `experimental_expose_trace_id` for a second and popping up the stack: I left a comment below that suggests renaming `experimental_expose_trace_id` to `experimental_response_trace_id`.  I think that _could_ cover the distinction about the response bit in the `header_name` property as well, and _also_ be more of an umbrella around the `enabled` thing because otherwise it isn't clear to me what `enabled: true` would be exposing without reading into optional properties inside it?

Reply:
For `custom_header`, I'll strawdog that maybe that could become `from_request_header: <string>`?

Reply:
thanks @abernix and @BrynCooke for suggestions, you're definitely better than me to find right names. I really like the proposal to call it `experimental_response_trace_id` and `from_request_header`. Just something to keep in mind, I think we should in the future rename `expose_query_plan` to `response_query_plan` or something that could be more consistent with this suggestion. But that's another topic. I will do this change thanks :)